### PR TITLE
feat: add mesh detection

### PR DESCRIPTION
**Pull request by pmalek**
This PR aims to bring mesh detection functionality from KIC tree. This mostly copies stuff from:

- https://github.com/Kong/kubernetes-ingress-controller/blob/dfe95422a337518e50b3f4618c14c50f29e94acc/internal/util/reports_mesh.go
- https://github.com/Kong/kubernetes-ingress-controller/blob/dfe95422a337518e50b3f4618c14c50f29e94acc/internal/util/reports_mesh_test.go
- and https://github.com/Kong/kubernetes-ingress-controller/tree/771eaeaf7dcaaf15342cf273f302dc228af1c746/internal/meshdetect

and adapts it to work with the framework.

### CODE REVIEW

File: pkg/internal/meshdetect/reports_mesh_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		},
+	}
+
+	for _, tc := range testCases {
+		// t.Run runs function in separate goroutine, so we need to assign tc to a local variable.
+		tc := tc
+		t.Run(tc.caseName, func(t *testing.T) {
+			// serialized := serializeMeshDeploymentResults(tc.results)
+			// require.Equalf(t, tc.serializedStr, serialized,
+			// 	"case %s: serialized message should be the same as expected", tc.caseName)
```
Comment (czeslavo) [inline pkg/internal/meshdetect/reports_mesh_test.go:56]: We're not running any test here, is that correct? 🤔

File: pkg/internal/meshdetect/reports_mesh_test.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+			serializedStr: "",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.caseName, func(t *testing.T) {
+			// serialized := serializeMeshRunUnderResults(tc.results)
+			// require.Equalf(t, tc.serializedStr, serialized,
+			// 	"case %s: serialized message should be the same as expected", tc.caseName)
```
Comment (czeslavo) [inline pkg/internal/meshdetect/reports_mesh_test.go:121]: ditto

File: pkg/internal/meshdetect/reports_mesh_test.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+			serializedStr: "",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.caseName, func(t *testing.T) {
+			// serialized := serializeMeshServiceDistribution(tc.results)
+			// require.Equalf(t, tc.serializedStr, serialized,
+			// 	"case %s: serialized message should be the same as expected", tc.caseName)
```
Comment (czeslavo) [inline pkg/internal/meshdetect/reports_mesh_test.go:157]: ditto

File: pkg/telemetry/workflowsk8smeshdetect.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	// MeshDetectWorkflowName is the name assigned to mesh detect workflow.
+	MeshDetectWorkflowName = "mesh-detect"
+)
+
+// NewMeshDetectWorkflow TODO
+//
+// Exemplar report produced:
+//
+//	{
+//	}
```
Comment (czeslavo) [inline pkg/telemetry/workflowsk8smeshdetect.go:19]: Is this the real expected report produced? 🤔

File: pkg/internal/meshdetect/reports_mesh_test.go (hunk 1 above)
Comment (pmalek) [reply pkg/internal/meshdetect/reports_mesh_test.go]: Yikes! 😬 Fixed.

File: pkg/telemetry/workflowsk8smeshdetect.go (hunk 4 above)
Comment (pmalek) [reply pkg/telemetry/workflowsk8smeshdetect.go]: Added a real example of a report produced by this workflow.

**Review by czeslavo — APPROVED** [review]

**Pull request merged**